### ISSUE TITLE
Inspection Settings throws an exception

### ISSUE THREAD

**Issue by bclothier** [rubberduck-vba/Rubberduck#3677]
We are getting an exception in `CodeInspectionSettings.cs`, on method `GetSetting` on this line:

```
var proto = Convert.ChangeType(Activator.CreateInstance(inspectionType), inspectionType);
```

The exception is:
```
System.MissingMethodException occurred
  HResult=0x80131513
  Message=No parameterless constructor defined for this object.
  Source=mscorlib
  StackTrace:
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic)
   at System.Activator.CreateInstance(Type type)
   at Rubberduck.Settings.CodeInspectionSettings.GetSetting(Type inspectionType) in C:\GitHub\Rubberduck\Rubberduck\RetailCoder.VBE\Settings\CodeInspectionSettings.cs:line 55
```

In this particular case, the `MissingAnnotationArgumentInspection` inspection has this constructor:

```
public MissingAnnotationArgumentInspection(RubberduckParserState state)
            : base(state, CodeInspectionSeverity.Error) { }
```

Thus, we must pass in the state *and* the severity with the above. Those are not available in the calling code.

The exception is caught then swallowed by returning `null` for the settings. Thus, it isn't directly observed when running RD normally, however, it trips up the debugger which must be Shift+F11 to force it into the `catch` block.

I believe this may be a side-effect of PR #3534 which changed how we stored the settings.

**Comment by bclothier** [comment]
Addendum - those inspections are what throws exceptions:

```
MissingAnnotationArgumentInspection
MissingAttributeInspection
BooleanAssignedInIfElseInspection
EmptyStringLiteralInspection
IllegalAnnotationInspection
ImplicitByRefModifierInspection
ModuleScopeDimKeywordInspection
MultilineParameterInspection
MultipleDeclarationsInspection
ObsoleteCallStatementInspection
ObsoleteCommentSyntaxInspection
ObsoleteErrorSyntaxInspection
ObsoleteLetStatementInspection
OptionBaseInspection
OptionExplicitInspection
ProcedureCanBeWrittenAsFunctionInspection
RedundantOptionInspection
StopKeywordInspection
```

**Comment by Vogel612** [comment]
The PR that actually introduced that code is #2215 ... At some point the Inspections must have been refactored to take a ParserState as ctor argument and only with the changes to InspectionSettings this change surfaced.

As of now the "hack" mentioned in #3678 is a sufficient fix, though. 
I'd say we close this with #3875

**Comment by bclothier** [comment]
I think this should be closed, since we already merged the PR #3876 (which aforementioned #3875 was merged into).
